api/api.go

```diff
@@ -345,6 +358,26 @@ func handleAuditingFailure(w http.ResponseWriter, err error, logData log.Data) {
 	http.Error(w, "internal server error", http.StatusInternalServerError)
 }
 
+func auditActionFailure(ctx context.Context, auditedAction string, auditedResult string, err error, logData log.Data) {
```

Comment:
Is this function likely to be wanted to be used by other services?

Maybe it would be worth adding this to the go-ns library?

Reply:
I did consider adding to the audit package. However that will hold up this change plus the others I am trying to get through. I would be happy to implement the rest of the auditing with this here and then move this into go-ns afterwards.

Reply:
Sounds good to me